### PR TITLE
chore(ui) Rename dashboardsV2 to dashboards

### PR DESCRIPTION
**Pull request by markstory**
We haven't had dashboards v1 code for many months now. Renaming these modules shouldn't cause any merge conflicts.

I did have to update a few `useEffect()` and `useMemo()` dependency arrays to resolve outstanding lint errors.

### CODE REVIEW

File: static/app/views/dashboards/create.tsx [hunk 1]
```diff
@@ -42,7 +42,8 @@ function CreateDashboard(props: Props) {
     if (constructedWidget) {
       browserHistory.replace(location.pathname);
     }
-  }, [location.pathname]);
+  }, [location.pathname, location.query]);
```
Comment (markstory) [inline static/app/views/dashboards/create.tsx:45]: The missing dep here was causing a lint error.

File: static/app/views/dashboards/view.tsx [hunk 2]
```diff
@@ -63,7 +63,7 @@ function ViewEditDashboard(props: Props) {
         query: pick(location.query, ALLOWED_PARAMS),
       });
     }
-  }, [location.pathname]);
+  }, [location.pathname, location.query]);
```
Comment (markstory) [inline static/app/views/dashboards/view.tsx:66]: The missing dep here was causing a lint error.

File: static/app/views/dashboards/widgetBuilder/buildSteps/groupByStep/groupBySelector.tsx [hunk 3]
```diff
@@ -84,7 +84,7 @@ export function GroupBySelector({fieldOptions, columns = [], onChange}: Props) {
         filteredFieldOptions: FieldOptions;
       }
     );
-  }, [fieldOptions, columns]);
+  }, [fieldOptions, columnFieldsAsString]);
```
Comment (markstory) [inline static/app/views/dashboards/widgetBuilder/buildSteps/groupByStep/groupBySelector.tsx:87]: The body of this memo doesn't use `columns` it uses `columnFieldsAsString` which is generated from `columns`.

File: static/app/views/dashboards/create.tsx (hunk 1 above)
Comment (markstory) [reply static/app/views/dashboards/create.tsx:45]: I'll double check that.

Comment (markstory) [reply static/app/views/dashboards/create.tsx:45]: I did some manual testing and created new dashboards, added widgets to them and cloned dashboards and didn't encounter any redirection loops.

**Review by k-fish — APPROVED** [review]

**Pull request merged**